OutSystems/outsystems-maps
# ROU-3174: Auto zoom for polyline shape

This PR is for adding the new Optional Config AutoZoomOnShapes.

### What was done
* Add the options that allow AutoZoom to adjust to fit all shapes and markers displayed on the map.
   * Add the autoZoomOnShapes to the Map configuration.
   * Add the providerBounds getter to each shape.
   * Updated the Zoom.refreshZoom() to consider the shapes' bounds.
* Note: For Leaflet, this feature does not update dynamically.

### Test Steps
1. Go to a page with a Map that contains Shapes
2. Set the Zoom = AutoZoom and AutoZoomOnShapes = true
3. Check that the Map Zoom automatically fits all the Shapes in the Map

### Checklist
